app/src/main/java/org/mozilla/focus/search/SearchEngine.java

```diff
@@ -101,4 +130,79 @@ private String paramSubstitution(String template, String query) {
 
         return template;
     }
+
+    public static boolean addSearchEngine(SharedPreferences sharedPreferences, Context context, String engineName, String searchQuery) {
```

Comment:
SearchEngine is a simple model class / POJO. I wouldn't overload its meaning by adding additional static methods to it. Maybe this could be added to `SearchEngineManager` (or its own class; or the `Settings` helper class)?